Sources/Tracking/Tracking.swift

```diff
@@ -0,0 +1,146 @@
+
+        return _identifyRepository
+    }
+
+    // swiftlint:enable identifier_name
+
+    private let keyValueStorage: KeyValueStorage
+
+    /// testing init
+    internal init(customerIO: CustomerIO?, identifyRepository: IdentifyRepository?, keyValueStorage: KeyValueStorage) {
+        self._identifyRepository = identifyRepository
+        self.keyValueStorage = keyValueStorage
+        self.customerIO = customerIO ?? CustomerIO(siteId: "fake", apiKey: "fake", region: Region.EU)
+    }
+
+    /**
+     Create a new instance of the `Tracking` class.
+
+     - Parameters:
+       - customerIO: Instance of `CustomerIO` class.
+     */
+    public init(customerIO: CustomerIO) {
+        self.customerIO = customerIO
+        self.keyValueStorage = DICommon.shared.keyValueStorage
+    }
+
+    /**
+     Identify a customer (aka: Add or update a profile).
+
+     [Learn more](https://customer.io/docs/identifying-people/) about identifying a customer in Customer.io
+
+     Note: You can only identify 1 profile at a time in your SDK. If you call this function multiple times,
+     the previously identified profile will be removed. Only the latest identified customer is persisted.
+
+     - Parameters:
+       - identifier: ID you want to assign to the customer.
+         This value can be an internal ID that your system uses or an email address.
+         [Learn more](https://customer.io/docs/api/#operation/identify)
+       - onComplete: Asynchronous callback with `Result` of identifying a customer.
+         Check result to see if error or success.
+       - email: Optional email address you want to associate with a profile.
+         If you use an email address as the `identifier` this is not needed.
+     */
+    public func identify(
+        identifier: String,
+        onComplete: @escaping (Result<Void, CustomerIOError>) -> Void,
+        email: String? = nil
+    ) {
+        guard let identifyRepository = self.identifyRepository else {
+            return onComplete(Result.failure(.notInitialized))
+        }
+
+        identifyRepository.addOrUpdateCustomer(identifier: identifier, email: email) { [weak self] result in
+            guard self != nil else { return }
+
+            switch result {
+            case .success:
+                return onComplete(Result.success(()))
+            case .failure(let error):
+                return onComplete(Result.failure(error))
+            }
+        }
+    }
+
+    /**
+     Stop identifying the currently persisted customer. All future calls to the SDK will no longer
+     be associated with the previously identified customer.
+
+     Note: If you simply want to identify a *new* customer, this function call is optional. Simply
+     call `identify()` again to identify the new customer profile over the existing.
+
+     If no profile has been identified yet, this function will ignore your request.
+     */
+    public func identifyStop() {
```

Comment:
wonder if we could come up with a better func name here, `identifyStop` feels not super intuitive